src/utils.ts

```diff
@@ -7,3 +7,6 @@ export function toHex (value: number, length: number = 2) {
 export function toBin (value: number, length: number = 8) {
     return value.toString(2).padStart(length, "0");
 }
+export function toDistance(value: number, partial: number = 1) {
+    return Math.floor((value + 1 / Math.max(partial, 1)) * 25.4) - 20;
```

Comment:
Note this changes the result when `partial` is 0. In the old code nothing was added to the inch distance, now you add 1. I am not sure what is the source for the previous formula (I am guessing empirical reverse engineering). At least after the change the result is not negative as often.

Reply:
I was just trying to make it a nice oneliner but I missed that is add 25.4 to the final result. It makes the result >5.4 in any case.

I did not understood that `partial` value so I tried to get more information. According to a [comment on  eurobricks](https://www.eurobricks.com/forum/index.php?/forums/topic/169010-boost-color-and-distance-sensor-readings/&tab=comments#comment-3051825) from @dlech, the format would be:

`<COLOR CODE> <DISTANCE> <LED COLOR> <REFLECTED LIGHT>`

If this right, the result should differ by ~24.5 for obstacles with significant reflectivity gap at the same distance.

It also explain why that `partial` is not part of the distance mode data. So SPEC_1 and PROX mode does not get the same accuracy in this implementation. I guess the formula should be removed for consistency ?